astropy/io/ascii/cparser.pyx

```diff
@@ -209,9 +210,19 @@ cdef class CParser:
             fast_reader = {}
         elif fast_reader is False: # shouldn't happen
             raise core.ParameterError("fast_reader cannot be False for fast readers")
-        # parallel and use_fast_reader are False by default
-        use_fast_converter = fast_reader.pop('use_fast_converter', False)
+        expchar = fast_reader.pop('exponent_style', 'E').upper()
+        # parallel and use_fast_reader are False by default, but only the latter
+        # supports Fortran double precision notation 
+        if expchar == 'E':
+            use_fast_converter = fast_reader.pop('use_fast_converter', False)
+        else:
+            use_fast_converter = fast_reader.pop('use_fast_converter', True)
+            if not use_fast_converter:
+                raise core.FastOptionsError("fast_reader: exponent_style requires use_fast_converter")
+            if expchar.startswith('FORT'):
```

Comment:
The documentation states that the parameter should be "Fortran", so check for equality to "FORTRAN" here.